cloudinit/sources/helpers/openstack.py

```diff
@@ -674,7 +674,9 @@ def convert_net_json(network_json=None, known_macs=None):
             if not mac:
                 raise ValueError("No mac_address or name entry for %s" % d)
             if mac not in known_macs:
-                raise ValueError("Unable to find a system nic for %s" % d)
+                LOG.error("Unable to find a system nic for %s. Skipping..." % d)
+                config.remove(d)
+                continue
```

Comment:
As it is, this is simply not OK. swallowing this error isn't really helping anyone.
The cloud platform told us there was a NIC that needed to be configured, if cloud-init merely logs the error, other things will break later and the user have to search the logs for "what went wrong".

possibly you could add a config option to cloud-init to "skip missing nics" or something, but the default behavior should be to fail.

Reply:
@smoser that makes sense to me. I will revise the patch to include a config parameter. Do you have any reference patch which adds a config to the cloud-init project?
P.S I am new to cloud-init code :)